certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -867,3 +875,18 @@ def install_ssl_options_conf(options_ssl):
     # Check to make sure options-ssl.conf is installed
     if not os.path.isfile(options_ssl):
         shutil.copyfile(constants.MOD_SSL_CONF_SRC, options_ssl)
+        return
+    # there's already a file there. if it exactly matches a previous file hash,
+    # we can update it. otherwise, create a .new and print a warning.
+    digest = crypto_util.sha256sum(options_ssl)
+    if digest in PREVIOUS_SSL_OPTIONS_HASHES: # safe to update
+        shutil.copyfile(constants.MOD_SSL_CONF_SRC, options_ssl)
+    elif digest == CURRENT_SSL_OPTIONS_HASH: # already up to date
+        return
+    else: # not safe to update
+        new_filename = options_ssl + ".new"
+        shutil.copyfile(constants.MOD_SSL_CONF_SRC, new_filename)
+        # print warning
+        logger.warning("%s has been manually modified; updated ssl configuration options "
```

Comment:
> Not sure what you mean; this is what we already do. Do you mean after they've edited it, save a copy somewhere else?

I mean save a copy somewhere on disk whenever we write `options-ssl-nginx.conf`. This way we can tell what file we placed there and whether or not we really have a new version of the file that contains security updates.

> We could also do checking of certbot version -- only do things once per version. I hasn't wanted to do that because it can get messy, but if we don't always want to create .new, there's various ways we could work version number in.

I think a hash, copy of the file, or version number stored somewhere would mitigate this problem. I kind of like a hash best because it's not likely to change with every update but any of these approaches is fine.

In my opinion, ideally we'd like to:

1. Only create `.new` when we actually have a new versions of the file.
2. Make as much noise as we reasonably can about the issue until the user resolves it one way or another.